gossip/src/cluster_info.rs

```diff
@@ -1243,6 +1283,29 @@ impl ClusterInfo {
             .collect()
     }
 
+    /// Returns last-voted-fork-slots inserted since the given cursor.
+    /// Excludes entries from nodes with unkown or different shred version.
+    pub fn get_restart_last_voted_fork_slots(
+        &self,
+        cursor: &mut Cursor,
+    ) -> Vec<(EpochSlotsIndex, EpochSlots, Slot, Hash)> {
+        let self_shred_version = Some(self.my_shred_version());
+        let gossip_crds = self.gossip.crds.read().unwrap();
+        gossip_crds
+            .get_restart_last_voted_fork_slots(cursor)
+            .filter(|entry| {
+                let origin = entry.value.pubkey();
+                gossip_crds.get_shred_version(&origin) == self_shred_version
+            })
+            .map(|entry| match &entry.value.data {
+                CrdsData::RestartLastVotedForkSlots(index, slots, last_vote, last_vote_hash) => {
+                    (*index, slots.clone(), *last_vote, *last_vote_hash)
```

Comment:
A tuple with 4 fields is not ideal.
I don't think you would need to expose `EpochSlotsIndex` outside of gossip code.
You probably also need to define a struct for `RestartLastVotedForkSlots`.

Reply:
Changed.